scim/resource_service_principal.go

```diff
@@ -63,8 +68,8 @@ func (a ServicePrincipalsAPI) Patch(servicePrincipalID string, r patchRequest) e
 }
 
 // Update replaces resource-friendly-entity
-func (a ServicePrincipalsAPI) Update(servicePrincipalID string, updateRequest User) error {
-	servicePrincipal, err := a.Read(servicePrincipalID)
+func (a ServicePrincipalsAPI) Update(servicePrincipalID string, attributes string, updateRequest User) error {
+	servicePrincipal, err := a.Read(servicePrincipalID, attributes)
```

Comment:
This method depends on the `groups` response, but that isn't part of `userAttributes`. Can you try the following (or whatever attribute should be present to get group membership info back)?
```suggestion
func (a ServicePrincipalsAPI) Update(servicePrincipalID string, updateRequest User) error {
	servicePrincipal, err := a.Read(servicePrincipalID, "groups")
```